tools/m2kcli/main.cpp

```diff
@@ -112,34 +114,37 @@ int main(int argc, char **argv)
 			return 0;
 		}
 
-		if (std::string(argv[1]) == "analog-in") {
-			auto *analogIn = new libm2k::cli::AnalogIn(argc, argv);
-			command = analogIn;
-		} else if (std::string(argv[1]) == "analog-out") {
-			auto *analogOut = new libm2k::cli::AnalogOut(argc, argv);
-			command = analogOut;
-		} else if (std::string(argv[1]) == "digital") {
-			auto *digital = new libm2k::cli::Digital(argc, argv);
-			command = digital;
-		} else if (std::string(argv[1]) == "power-supply") {
-			auto *powerSupply = new libm2k::cli::PowerSupply(argc, argv);
-			command = powerSupply;
-		} else if (std::string(argv[1]) == "spi") {
-			auto *spi = new libm2k::cli::Spi(argc, argv);
-			command = spi;
-		} else if (std::string(argv[1]) == "i2c") {
-			auto *i2c = new libm2k::cli::I2c(argc, argv);
-			command = i2c;
-		} else if (std::string(argv[1]) == "uart") {
-			auto *uart = new libm2k::cli::Uart(argc, argv);
-			command = uart;
-		} else if (std::string(argv[1]) == "uart-terminal") {
-			auto *uartTerminal = new libm2k::cli::UartTerminal(argc, argv);
-			command = uartTerminal;
-		} else {
-			throw std::runtime_error("m2kcli: '" + std::string(argv[1]) +
-						 "' is not a m2kcli command. See 'm2kcli --help'.\n");
-		}
+                if (std::string(argv[1]) == "context") {
+                    libm2k::cli::Context Context(argc, argv);
```

Comment:
Should be pointer, otherwise the object won't be polymorphic and the method called (parseArguments) will be the base one - pure virtual method -> runtime error.